src/python/zquantum/qaoa/problems/problem.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+import networkx as nx
+from openfermion import QubitOperator
+from typing import Tuple, List
+from ._problem_evaluation import (
+    evaluate_solution,
+    solve_graph_problem_by_exhaustive_search,
+)
+
+
+class Problem(ABC):
+    @staticmethod
+    @abstractmethod
+    def build_hamiltonian(graph: nx.Graph, kwargs):
+        """Nothing to do here"""
```

Comment:
Isn't `...` a better way to indicate this than docstring?

Reply:
I tried using `pass` and it didn't work. Aren't the dots equivalent to pass?

Reply:
I'm worried about mixing `@staticmethod` with `@abstractmethod`. It certainly is confusing to me, it may also be causing problems like this one where `pass` as the method body wouldn't work.

Reply:
Right, there are separate decorators for that:
- https://docs.python.org/3/library/abc.html#abc.abstractclassmethod
- https://docs.python.org/3/library/abc.html#abc.abstractstaticmethod

I suggest we go with instance methods anyway 😄 more info on Zoom, @Shiro-Raven